.github/workflows/release.yml

```diff
@@ -1,39 +1,75 @@
-name: Publish Gem
+name: Release
 
 on:
   push:
-    tags:
-      - v*
+    branches:
+      - "master"
 
 jobs:
-  test:
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu, macos]
-        ruby: [2.6, 2.7, 3.1, truffleruby]
-    runs-on: ${{ matrix.os }}-latest
-    continue-on-error: ${{ endsWith(matrix.ruby, 'head') || matrix.ruby == 'debug' }}
+  run-tests:
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: ruby/setup-ruby@v1
+      - name: Checkout code
+        uses: actions/checkout@v2
         with:
-          ruby-version: ${{ matrix.ruby }}
-      - run: bundle install
-      - run: bundle exec rake
+          fetch-depth: 0
+          token: ${{ secrets.GH_TOKEN }}
 
-  build:
-    runs-on: ubuntu-latest
-    needs: test
-    
-    steps:
-      - uses: actions/checkout@v2
-      - uses: ruby/setup-ruby@v1
+      - name: Get tags
+        run: git fetch --tags origin
+  
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Setup Poetry
+        uses: abatilo/actions-poetry@v2.1.2
+        with:
+          poetry-version: 1.4.2
```

Comment:
We should be able to remove these right?

Reply:
1000%